run.sh

```diff
@@ -114,6 +114,23 @@ fi
 # configuration overrides
 # CONF directory and files need to be writable by the user running the container
 
+## GC Log Settings
+if [[ ! -z ${ES_GCLOG_FILE_COUNT} ]]; then
```

Comment:
_pedantry:_ I think shellcheck would complain about this and suggest `[[ -n "${ES_GCLOG_FILE_COUNT}" ]]` (there's a [pycharm plugin](https://plugins.jetbrains.com/plugin/10195-shellcheck) and shellcheck can be installed using `update-tool`/companion kube

Reply:
in our case though ES_GCLOG_FILE_COUNT is always initialized since we have a default.

if "someone" though was going to set it to "" ( empty ) in a kubernetes yaml, would that sitll match with [[ -n ]] ? 

-z would match that

Reply:
`[[ -n ]]` will fail, whereas `[[ -n "" ]]` won't match. I think with bash to detect unset vs. empty you need to do `[ -n "${var-}" ] && [ "${var-x}" = x ]` (the first `${var-}` is just to work with `set -o nounset`)

Reply:
Unless this is a real issue i would merge it as it is. All the other checks we do are "-z" can you in case create a separate case for this change of "bash variable checks" ?

Reply:
no complaints here, we don't have linting in the lifecycle of this